### PR TITLE
💻 format all old adventure contents with curlies

### LINKED ISSUE #5310: 💻 old adventures still have curlies
In old adventures, if you edit the existing code (which had curlies) they now show:

image

_Originally posted by @Felienne in https://github.com/hedyorg/hedy/issues/5297#issuecomment-2018345922_

### PR DESCRIPTION
**Pull request by hasan-sh**
Fixes #5310 

**How to test?**
- have an existing adventure with curly braces
- try to see this adventure from different locations; /hedy, /public-adventures, etc.

### CODE REVIEW

File: website/frontend_types.py [hunk 1]
```diff
@@ -143,7 +143,8 @@ def get(self, key, default=None):
 
     @staticmethod
     def from_teacher_adventure_database_row(row):
-        text, example_code = halve_adventure_content(row["content"])
+        text, example_code = halve_adventure_content(
+            row["formatted_content"] if row.get("formatted_content") else row["content"])
```
Comment (Felienne) [inline website/frontend_types.py:147]: this work but did you know that you can supply get with a default argument too? This code does the same:

`row.get("formatted_content", row["content"])`

(See: https://docs.python.org/3/library/stdtypes.html#dict.get)

Comment (hasan-sh) [reply website/frontend_types.py:147]: Sure, i use this elsewhere as well, this commit i rolled back from another pr. WIll change it now.

**Comment by Felienne** [comment]
The curlies issue is solved by somehow I now again don't see all newlines:

image

(I have updated from main)

**Comment by hasan-sh** [comment]
> The curlies issue is solved by somehow I now again don't see all newlines:
> 
> (I have updated from main)

Where is it resolved? because i still see it on alpha!
image


Perhaps what you see is an adventure that's been saved recently?

**Comment by Felienne** [comment]
Haha, it looks differently weird now, the newlines are there but there is a weird strip above?

image

But I guess the newlines indeed is gone now! Let me know if you still need the local json do dive into this line.

File: website/public_adventures.py [hunk 2]
```diff
@@ -49,32 +50,35 @@ def init(self, user):
...
-                    "cloned_times": adventure.get("cloned_times"),
-                    "tags": adv_tags,
-                    "text": adventure.get("content"),
-                    "is_teacher_adventure": True,
-                    # "content": adventure.get("content")
-                }
-            )
+
+            content = safe_format(adventure['formatted_content']
+                                  if adventure.get("formatted_content") else adventure["content"],
```
Comment (jpelay) [inline website/public_adventures.py:55]: Can we use the same `get` as you did in the other file here? Looks a lot cleaner.

```suggestion
            content = safe_format(adventure.get('formatted_content', adventure['content']),
```

File: website/public_adventures.py [hunk 3]
```diff
@@ -49,32 +50,35 @@ def init(self, user):
...
+                "creator_public_profile": public_profile,
+                "date": utils.localized_date_format(adventure.get("date")),
+                "level": adventure.get("level"),
+                "levels": adventure.get("levels"),
+                "language": adv_lang,
+                "cloned_times": adventure.get("cloned_times"),
+                "tags": adv_tags,
+                "text": content,
+                "is_teacher_adventure": True,
+                # "content": adventure.get("content")
```
Comment (jpelay) [inline website/public_adventures.py:72]: We can also delete this comment here.

**Review by jpelay — CHANGES_REQUESTED** [review]
Works great, just made a couple of comments to make the code tidier, but otherwise looks good to go.

**Review by Felienne — APPROVED** [review]

**Pull request merged**